src/_pytest/mark/expression.py

```diff
@@ -5,7 +5,8 @@
 expression: expr? EOF
 expr:       and_expr ('or' and_expr)*
 and_expr:   not_expr ('and' not_expr)*
-not_expr:   'not' not_expr | '(' expr ')' | ident
+not_expr:   'not' not_expr | '(' expr ')' | ident ( '(' name '=' value ( ', ' name '=' value )*  ')')*
```

Comment:
- Would remove the space after `ident (` for consistency.
- The `*` (zero-or-more) at the end of this line should be `?` (zero-or-one), I think.
- The definitions of the `name` and `value` productions are missing.